src/primitives/transaction.h

```diff
@@ -432,10 +434,10 @@ class CTxIn
                 fHasAssetIssuance = false;
```

Comment:
is it OK that ``prevout.n`` (and later ``output.n``) has the ``COutPoint::OUTPOINT_PEGIN_FLAG`` set here? even if ``m_is_pegin`` is false? Perhaps there's some invariant I'm not aware of, but if ``m_is_pegin`` is always set in this branch of the if, perhaps there should be an ``assert()`` to indicate that

Reply:
this is a special casing for coinbase inputs. Can you point to the behavior you're worried about exactly? In both directions this case is handled.

Reply:
I get the impression there's an invariant here along the lines of ``m_is_pegin`` <=> ``output.n & COutPoint::OUTPOINT_PEGIN_FLAG``.

But in the (top) if-block (line 431 on the right), this invariant does not seem to be maintained, since ``prevout.n`` can have all bits set, regardless of what ``m_is_pegin`` is. i.e., that branch of the if-statement allows serializing the ``OUTPOINT_PEGIN_FLAG`` regardless.

Perhaps ``m_is_pegin`` is always true if ``prevout.n == -1``. If so, perhaps it would make sense to add an assert to indicate that.

Reply:
Right, it's another magic value for coinbase input values, which also effects issuance bit logic.

Perhaps more rigorous asserts are needed; any specific places to suggest?